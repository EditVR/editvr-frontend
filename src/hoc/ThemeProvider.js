/**
 * @file ThemeProvider.js
 * Exports a HOC ThemeProvider component that attaches children their MUI theme.
 */

import React from 'react';
import { MuiThemeProvider } from '@material-ui/core';

import muiTheme from '../styles/muiTheme';

const ThemeProvider = WrappedComponent => props => (
  <MuiThemeProvider theme={muiTheme}>
    <WrappedComponent {...props} />
  </MuiThemeProvider>
);

export default ThemeProvider;
