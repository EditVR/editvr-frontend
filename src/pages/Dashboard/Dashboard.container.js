/**
 * @file Dashboard.container.js
 * Exports a redux-connected Dashboard component.
 */

import { connect } from 'react-redux';

import Dashboard from './Dashboard';

const mapDispatchToProps = dispatch => ({
  dispatch
});

const mapState = ({ user, experiences }) => ({
  user,
  experiences
});

export default connect(mapState, mapDispatchToProps)(Dashboard);
