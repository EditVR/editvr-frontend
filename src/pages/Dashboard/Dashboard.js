/**
 * @file Dashboard.js
 * Exports a React component that render's EditVR's editorial dashboard.
 */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import {
  Typography,
  Button,
  Card,
  CardContent,
  CardActions,
  Tooltip,
  withStyles
} from '@material-ui/core';
import { AddBox, Edit, OpenInBrowser } from '@material-ui/icons';

import { DashboardLayout } from '../../layouts';
import { Message } from '../../components';
import { EXPERIENCES_FETCH_FOR_USER } from '../../constants';
import DashboardStyles from './Dashboard.style';

class Dashboard extends Component {
  static propTypes = {
    classes: PropTypes.shape({
      card: PropTypes.string.isRequired,
      buttons: PropTypes.string.isRequired,
      buttonIcon: PropTypes.string.isRequired
    }).isRequired,
    user: PropTypes.shape({
      uid: PropTypes.string.isRequired,
      authentication: PropTypes.shape({
        accessToken: PropTypes.string.isRequired,
        csrfToken: PropTypes.string.isRequired
      }).isRequired
    }).isRequired,
    experiences: PropTypes.shape({
      error: PropTypes.string,
      items: PropTypes.arrayOf(
        PropTypes.shape({
          title: PropTypes.string.isRequired,
          body: PropTypes.shape({
            value: PropTypes.string
          }),
          field_experience_path: PropTypes.string
        })
      )
    }),
    dispatch: PropTypes.func.isRequired
  };

  static defaultProps = {
    experiences: {
      error: null
    }
  };

  /**
   * {@inheretdoc}
   */
  componentWillMount() {
    const { dispatch, user } = this.props;
    dispatch({
      type: EXPERIENCES_FETCH_FOR_USER,
      user
    });
  }

  /**
   * {@inheretdoc}
   */
  render() {
    const {
      classes,
      experiences: { error, items }
    } = this.props;

    return (
      <DashboardLayout title="Experiences">
        <Typography component="p">
          On this page you can create an experience, or open one of your
          existing experiences for editing by clicking the Open button.
        </Typography>
        <div className={classes.buttons}>
          <Button
            variant="raised"
            color="primary"
            component={Link}
            to="/experience/create"
          >
            Create
            <AddBox className={classes.buttonIcon} />
          </Button>
        </div>
        {error && <Message>{error}</Message>}
        {Object.entries(items).map(
          ([key, { title, body, field_experience_path: path }]) => (
            <Card key={key} className={classes.card}>
              <CardContent>
                <Typography gutterBottom variant="headline">
                  {title}
                </Typography>
                <Typography
                  component="p"
                  dangerouslySetInnerHTML={{
                    __html: body
                      ? body.value
                      : 'This experience does not yet have a description.'
                  }}
                />
              </CardContent>
              <CardActions>
                <Tooltip title={`Open ${title} in VR editor`}>
                  <Button
                    variant="outlined"
                    size="small"
                    component={Link}
                    className={classes.cardActionButton}
                    to={`/experience/editvr/${path}`}
                  >
                    <OpenInBrowser />
                  </Button>
                </Tooltip>
                <Tooltip title={`Configure ${title}`}>
                  <Button
                    variant="outlined"
                    size="small"
                    component={Link}
                    className={classes.cardActionButton}
                    to={`/experience/edit/${path}`}
                  >
                    <Edit />
                  </Button>
                </Tooltip>
              </CardActions>
            </Card>
          )
        )}
      </DashboardLayout>
    );
  }
}

export default withStyles(DashboardStyles)(Dashboard);
