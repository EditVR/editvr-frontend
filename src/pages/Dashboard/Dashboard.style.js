/**
 * @file Dashboard.style.js
 * Exports Dashboard page styles.
 */

export default theme => ({
  card: {
    marginTop: theme.spacing.unit * 2
  }
});
