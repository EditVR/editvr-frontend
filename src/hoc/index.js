/**
 * @file index.js
 * Exports all hoc components.
 */

import ThemeProvider from './ThemeProvider';

export { ThemeProvider };
