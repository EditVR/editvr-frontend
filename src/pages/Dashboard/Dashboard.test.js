/**
 * @file Dashboard.test.js
 * Contains tests for Dashboard.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';

import { ThemeProvider } from '../../hoc';
import Dashboard from './Dashboard.container';

const ThemedDashboard = ThemeProvider(Dashboard);

describe('<Dashboard />', () => {
  it('Matches its snapshot', () => {
    const store = configureStore()({
      user: {
        uid: '1',
        authentication: {
          accessToken: 'test',
          csrfToken: 'test'
        }
      },
      experiences: {
        loading: false,
        error: null,
        items: [
          {
            title: 'test experience',
            body: {
              value: 'this is a test experience'
            },
            field_experience_path: 'test'
          }
        ]
      },
      loadingBar: {
        default: 0
      }
    });
    expect(
      renderer
        .create(
          <Provider store={store}>
            <Router>
              <ThemedDashboard />
            </Router>
          </Provider>
        )
        .toJSON()
    ).toMatchSnapshot();
  });
});
