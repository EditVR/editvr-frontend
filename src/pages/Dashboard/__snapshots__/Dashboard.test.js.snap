// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> Matches its snapshot 1`] = `
<div
  className="ThinLayout-wrapper-2"
  id="layout__wrapper"
>
  <div />
  <div
    className="ThinLayout-thinWrapper-3"
    id="layout__thin-wrapper"
  >
    <header
      className="Header-wrapper-4"
    >
      <img
        alt="EditVR logo"
        className="Header-logo-5"
        src="editvr-logo.svg"
      />
    </header>
    <h1
      className="MuiTypography-root-6 MuiTypography-headline-11"
    >
      Experiences
    </h1>
    <p
      className="MuiTypography-root-6 MuiTypography-body1-15"
    >
      Please select an experience below. Click the Open button to open the experience for editing.
    </p>
    <div
      className="MuiPaper-root-31 MuiPaper-elevation2-35 MuiPaper-rounded-32 MuiCard-root-30 Dashboard-card-1"
    >
      <div
        className="MuiCardContent-root-58"
      >
        <h1
          className="MuiTypography-root-6 MuiTypography-headline-11 MuiTypography-gutterBottom-23"
        >
          test experience
        </h1>
        <p
          className="MuiTypography-root-6 MuiTypography-body1-15"
          dangerouslySetInnerHTML={
            Object {
              "__html": "this is a test experience",
            }
          }
        />
      </div>
      <div
        className="MuiCardActions-root-59"
      >
        <a
          className="MuiButtonBase-root-77 MuiButton-root-61 MuiButton-flatPrimary-63 MuiButton-sizeSmall-74 MuiCardActions-action-60"
          href="/edit/test"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-62"
          >
            Open
          </span>
          <span
            className="MuiTouchRipple-root-80"
          />
        </a>
      </div>
    </div>
    <footer
      className="Footer-wrapper-87"
    >
      <div
        className="Footer-elements-88"
      >
        <a
          href="https://fourkitchens.com"
        >
          <p
            className="MuiTypography-root-6 MuiTypography-body1-15 Footer-text-90"
            type="body1"
          >
            EditVR is a product of
          </p>
          <img
            alt="Four Kitchens"
            className="Footer-logo-89"
            src="4k-logo-reversed.svg"
          />
        </a>
      </div>
    </footer>
  </div>
</div>
`;
